include/libp2p/protocol/gossip/impl/ptr_key_wrapper.hpp

```diff
@@ -0,0 +1,124 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef LIBP2P_PTR_KEY_WRAPPER_HPP
+#define LIBP2P_PTR_KEY_WRAPPER_HPP
+
+#include <memory>
+
+namespace libp2p {
+
+  /**
+   * Helper to use smart pointers as a key in associative containers
+   *
+   * e.g.
+   * std::multiset<UniquePtrKeyWrapper<SomeType>, std::less<>> s;
+   *   s.emplace(std::make_unique<SomeType>(...));
+   *   SomeType other(...);
+   *   s.find(other);
+   */
+  template <typename PtrType>
+  struct PtrKeyWrapper {
```

Comment:
Why complicate that much with such wrapper? Shared_ptr itself is comparable as it has operator < (https://en.cppreference.com/w/cpp/memory/shared_ptr/operator_cmp)
Please remove this file and define set of peers as `std::set<shared_ptr<PeerContext>>` as this is the only place where you really use this method.

Reply:
If order of peers matter then only redefine less operator for shared_ptr<PeerContext>